core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -177,6 +179,14 @@ def is_valid(self) -> bool:
         """
         return self.is_fit()
 
+    def is_initialized(self) -> bool:
```

Comment:
This is reasonable, but the other property candidate methods such as `is_valid()` would be API breaking if we switched them to properties. We may consider switching all of them to properties in future, but it will be API breaking.

Reply:
Shall we create an issue to track this?

Thoughts (not for this PR): This is 'internal' APIs - only model developers will be impacted. Saying this, we probably want a migration guide for the change if we decide to go with the properties in future.

For properties `is_` prefix probably not needed: `valid` vs `is_valid`

Reply:
Probably doesn't need its own issue ATM. If we are to API break AbstractModel, we'd probably want to do a lot of API-break related changes at once. If we do we would of course want to have a migration guide.